aep/general/0127/aep.md.j2

```diff
@@ -1,5 +1,132 @@
 # HTTP and gRPC Transcoding
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/51) for more
-information.
+APIs that follow [resource-oriented design][resources] are defined using
+[RPCs][rpc], but the resource-oriented design framework allows them to also be
+presented as APIs that largely follow REST/JSON conventions. This is important
+in order to help developers use their existing knowledge: over 80% of the
+public APIs available follow most REST conventions, and developers are
+accustomed to that pattern.
+
+## Guidance
+
+APIs **must** provide HTTP definitions for each RPC that they define, except
+for bi-directional streaming RPCs, which can not be natively supported using
+HTTP/1.1. When providing a bi-directional streaming method, an API **should**
+also offer an alternative method that does not rely on bi-directional
+streaming.
+
+### HTTP method and path
+
+When using protocol buffers, each RPC **must** define the HTTP method and path
+using the `google.api.http` annotation:
+
+```proto
+rpc CreateBook(CreateBookRequest) returns (Book) {
+  option (google.api.http) = {
+    post: "/v1/{parent=publishers/*}/books"
+    body: "book"
+  };
+}
+
+message CreateBookRequest {
+  // The publisher who will publish this book.
+  // When using HTTP/JSON, this field is automatically populated based
+  // on the URI, because of the `{parent=publishers/*}` syntax.
+  string parent = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      child_type: "library.googleapis.com/Book"
+    }];
+
+  // The book to create.
+  // When using HTTP/JSON, this field is populated based on the HTTP body,
+  // because of the `body: "book"` syntax.
+  Book book = 2 [(google.api.field_behavior) = REQUIRED];
+
+  // The user-specified ID for the book.
+  // When using HTTP/JSON, this field is populated based on a query string
+  // argument, such as `?bookId=foo`. This is the fallback for fields that
+  // are not included in either the URI or the body.
+  // Note that clients use camelCase format to communicate the field names
+  // to the service.
+  string book_id = 3;
+}
+```
+
+- The first key (`post` in this example) corresponds to the HTTP method. RPCs
+  **may** use `get`, `post`, `patch`, or `delete`.
+  - RPCs **must** use the prescribed HTTP verb for each standard method, as
+    discussed in [Get][/get], [List][/list], [Create][/create], [Update][/update], and
```

Comment:
Does this link syntax work (e.g. `[Get][/get]`)?  I think we use `[Get](/get)` elsewhere.

I'd expect `[Get][get]` to work because `[get]` is defined below.  But we've been doing more of the inlined `[Get](./get)` style recently.

Reply:
I changed the style to the only style I could get to work ([Get][]). I've seen some others with this style as well. Let me know if you want to change it to something else.

Reply:
Here's an example of the style I mean:  https://github.com/aep-dev/aep.dev/blob/main/aep/general/0148/aep.md.j2#L23

Note that there's no link definition like `[list]` or `[create]` at the bottom of the document.